bindings/protocols/coap/index.html

```diff
@@ -61,79 +61,130 @@ <h2>Protocol Overview</h2>
-                    <td><code>SecurityScheme</code></td>
-                </tr>
-                -->
-            </tbody>
-        </table>
+        <section>
+            <h3>Requests</h3>
+            <p>
+                ... <code>href</code> and <code>coap:method</code>.
+            </p>
+            <pre id="example-feature-request" class="example" title="Requests">
+{
+    "op": [ "readproperty" ],
+    "href": "coap://[2001:DB8::1]/.../...",
+    "coap:method": "GET"
+}
+</pre>
+            <p>
+                ... <code>coap:format</code>.
+            </p>
+            <pre id="example-feature-format" class="example" title="Payload Format">
+{
+    "op": [ "writeproperty" ],
+    "href": "coap://[2001:DB8::1]/.../...",
+    "coap:method": "PUT",
+    "coap:format": 112  // application/senml+cbor
+}
+</pre>
+        </section>
+        <section>
+            <h3>Content Negotiation</h3>
+            <p>
+                ... <code>coap:accept</code>.
+            </p>
+            <pre id="example-feature-accept" class="example" title="Content Negotiation">
+{
+    "op": [ "readproperty" ],
+    "href": "coap://[2001:DB8::1]/.../...",
+    "coap:method": "GET",
+    "coap:accept": 112  // application/senml+cbor
+}
+</pre>
+        </section>
+        <section>
+            <h3>Caching</h3>
+            <p>
+                ...
+            </p>
+        </section>
+        <section>
+            <h3>Observing Resources</h3>
+            <p>
+                Observing resources as defined in [[RFC7641]] can be enabled by including a
+                <code>subprotocol</code> value of <code>coap:observe</code> in a form.
+            </p>
+            <pre id="example-feature-observe" class="example" title="Observing Resources">
+{
+    "op": [ "observeproperty" ],
+    "href": "coap://[2001:DB8::1]/.../...",
+    "coap:method": "GET",
+    "subprotocol": "coap:observe"
+}
+</pre>
+        </section>
+        <section>
+            <h3>Block-Wise Transfers</h3>
+            <p>
+                Parameters for blockwise transfer (as defined in [[RFC7959]]) can be set in a
+                <code>coap:blockwise</code> member.
+            </p>
+            <p>
+                ... <code>coap:blockwise</code> and <code>coap:quickblockwise</code>.
+            </p>
+            <pre id="example-feature-blockwise" class="example" title="Block-Wise Transfer">
+{
+    "op": [ "readproperty" ],
+    "href": "coap://[2001:DB8::1]/.../...",
+    "coap:method": "GET",
+    "coap:blockwise": { }
+}
+</pre>
+            <pre id="example-feature-blockwise-parameters" class="example" title="Block Size">
+{
+    "op": [ "readproperty" ],
+    "href": "coap://[2001:DB8::1]/.../...",
+    "coap:method": "GET",
+    "coap:blockwise": {
+        "coap:block2SZX": 64
+    }
+}
+</pre>
+        </section>
+        <section>
+            <h3>Conditional Requests</h3>
+            <p>
+                ...
+            </p>
+        </section>
```

Comment:
On the other hand, servers that can't handle If-Match or If-None-Match options should probably reject them anyway?

>  For each of these options, if the condition given is not fulfilled,
   then the server MUST NOT perform the requested method.  Instead, the
   server MUST respond with the 4.12 (Precondition Failed) Response
   Code.
   – [RFC 7252, section 5.10.8](https://datatracker.ietf.org/doc/html/rfc7252#section-5.10.8)

Reply:
You raise a good point.

There are three variants of conditional requests:

* `If-Match` with an ETag: "only do this if the resource exists and has this representation"
* `If-Match` without an ETag: "only do this if the resource exists"
* `If-None-Match` without an ETag: "only do this if the resource does not exist"

I don't think we should put specific ETags in Thing Descriptions; I'm not sure about the rest... Usually, conditional requests are used in the context of some kind of larger workflows, where for example you'd want to avoid multiple clients making concurrent updates to a resource. Should we try to describe those workflows in a Thing Description? Should the client just "know" to use those workflows and we just put an indication that the server supports conditional requests in the Thing Description? I don't have a good idea yet.

Reply:
Hmm, could we maybe put these three approaches into the document and define some vocabulary to indicate that the affordance supports one or more of them? Or would this be too broad? If so, are there perhaps some well-known workflows we could reference?

If we don't find a good solution for this PR, we could consider creating an issue from this comment and revisit the problem later, I would say :)

Reply:
I think I'd describe the three approaches in this PR but leave any definition of vocabulary for these to a new issue.